### PR TITLE
Issue 68 footer copyright from cms

### PR DESCRIPTION
**Pull request by drewrogersuxd**
creates getFooterData so footer now pulls text/images from CMS

### CODE REVIEW

File: src/model/footer.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+export const zFooterData = z.object({
+    logo: zImage,
+    copyrightDate: z.coerce.date(),
+    copyrightNotice: z.string(),
+    organizationName: z.string(),
+    legalText: zPortableText,
+});
+
+export const gFooterDataQuery = groq`
+*[_id == "organization-singleton"][0]{
```
Comment (danlechambre) [inline src/model/footer.ts:17]: This looks like an invalid query - but it looks like you cracked it with the query you wrote inside the `client` method.

I'd recommend replacing this with the groq query you defined in  `getFooterData` (I've found they're easier to debug when you can view them side by side with the model that parses them). Then you can just import that into the client and pass into the fetch method

File: src/common/client.ts [hunk 2]
```diff
@@ -368,18 +373,29 @@ export async function getFallbackImageCollection(): Promise<SanityImage[]> {
...
- * @todo We probably want to give the footer it's own content type in the future
+ * Gets legal and copyright text for footer from org
+ *
+ * @returns
  */
-export async function getSmallPrint(): Promise<PortableText> {
+export async function getFooterData(): Promise<FooterData> {
   const client = getClient();
-  const query = groq`*[_id == "organization-singleton"][0].smallPrint`;
+  const query = groq`*[_id == "organization-singleton"][0]{
```
Comment (danlechambre) [inline src/common/client.ts:382]: See other comment - move this into `model/footer` then use directly in fetch method.

See `getAboutPageData` function below for example

File: src/ui/SiteFooter.astro
Comment (danlechambre) [inline src/ui/SiteFooter.astro:1]: Noticed a couple of formatting quirks in this PR so wondered if you have the Prettier extension installed in VS Code? 

There's a couple of prettier config files in the root of the repo that set the rules for the project.

I tend to have this setting turned on in my VS Code config:
CleanShot 2024-12-11 at 18 46 39@2x

And I also tend to set this:
CleanShot 2024-12-11 at 18 53 39@2x

Just means we never have to think about code formatting during PR 🙌

**Review by danlechambre — CHANGES_REQUESTED** [review]
Great work Drew. Just a couple of bits before we get this one over

File: src/ui/SiteFooter.astro
Comment (drewrogersuxd) [reply src/ui/SiteFooter.astro:1]: ok thanks @danlechambre.
I added Prettier extension and turned on the format on save
Hope it fixes the issues I introduced

File: src/common/client.ts (hunk 2 above)
Comment (drewrogersuxd) [reply src/common/client.ts:382]: ok thanks will do

File: src/model/footer.ts (hunk 1 above)
Comment (drewrogersuxd) [reply src/model/footer.ts]: @danlechambre Please review my latest fixes. I think it is the way that you recommended, now.
thanks for the help

**Review by danlechambre — APPROVED** [review]
Looks great Drew, changes look to be fine on Cloudflare too 🎉

**Comment by danlechambre** [comment]
Closes #68

**Pull request merged**